src/samples/Authorization/ManualSignIn/AuthAgent.cs

```diff
@@ -0,0 +1,217 @@
+        OnMessage("-signin", SignInAsync);
+
+        // Handles the user sending a Logout command using the specific keywords '-signout'
+        OnMessage("-signout", async (turnContext, turnState, cancellationToken) =>
+        {
+            // force a user signout to reset the user state
+            // this is needed to reset the token in Azure Bot Services if needed. 
+            await Authorization.SignOutUserAsync(turnContext, turnState, cancellationToken: cancellationToken);
+            await turnContext.SendActivityAsync("You have signed out", cancellationToken: cancellationToken);
+        }, rank: RouteRank.Last);
+
+
+        /*
+         The Authorization Class provides methods and properties to manage and access user authentication tokens
+         You can use this class to interact with the authentication process, including signing in and signing out users, accessing tokens, and handling authentication events.
+         */
+
+        // Register Events for SignIn and SignOut on the authentication class to track the status of the user, notify the user of the status of their authentication process, or log the status of the authentication process.
+        Authorization.OnUserSignInSuccess(OnUserSignInSuccess);
+
+        Authorization.OnUserSignInFailure(async (turnContext, turnState, handlerName, response, initiatingActivity, cancellationToken) =>
+        {
+            await turnContext.SendActivityAsync($"Manual Sign In: Failed to login to '{handlerName}': {response.Error.Message}", cancellationToken: cancellationToken);
+        });
+
+        // Registers a general event handler that will pick up any message activity that is not covered by the previous events handlers. 
+        OnActivity(ActivityTypes.Message, OnMessageAsync, rank: RouteRank.Last);
+    }
+
+    /// <summary>
+    /// This method is called to handle the conversation update event when a new member is added to or removed from the conversation.
+    /// </summary>
+    /// <param name="turnContext"><see cref="ITurnContext"/></param>
+    /// <param name="turnState"><see cref="ITurnState"/></param>
+    /// <param name="cancellationToken"><see cref="CancellationToken"/></param>
+    private async Task WelcomeMessageAsync(ITurnContext turnContext, ITurnState turnState, CancellationToken cancellationToken)
+    {
+        /*
+        In this example, we will send a welcome message to the user when they join the conversation.
+        We do this by iterating over the incoming activity members added to the conversation and checking if the member is not the agent itself.
+        Then a greeting notice is provided to each new member of the conversation.
+        */
+        foreach (ChannelAccount member in turnContext.Activity.MembersAdded)
+        {
+            if (member.Id != turnContext.Activity.Recipient.Id)
+            {
+                StringBuilder sb = new StringBuilder();
```

Comment:
nit, use `"""` literals for easy mainteinance (dont think StringBuilder is needed)

Reply:
fair.. accepted